packages/tests/src.ts/test-contract.spec.ts

```diff
@@ -463,11 +314,73 @@ describe("Test Contract Transaction Population", function() {
    }).timeout(60000);
 });
 
-describe("contract.deployed", function() {
-    const hederaEoa = {
-        account: '0.0.29562194',
-        privateKey: '0x3b6cd41ded6986add931390d5d3efa0bb2b311a8415cfe66716cac0234de035d'
+describe('Contract Events', function () {
+    const provider = ethers.providers.getDefaultProvider('testnet');
+    // @ts-ignore
+    const wallet = new ethers.Wallet(hederaEoa, provider);
+    const abiGLDTokenWithConstructorArgs = JSON.parse(readFileSync('examples/assets/abi/GLDTokenWithConstructorArgs_abi.json').toString());
+    const contract = ethers.ContractFactory.getContract('0x0000000000000000000000000000000001c42805', abiGLDTokenWithConstructorArgs, wallet);
+    const sleep = async (timeout: number) => {
+        await new Promise(resolve => {
+            console.log(`Waiting ${timeout/1000} secs before removing contract listeners.`)
+            setTimeout(resolve, timeout);
+        });
     };
+
+    it("should be able to capture events via contract", async function () {
+        const capturedMints: any[] = [];
+        contract.on('Mint', (...args) => {
+            assert.strictEqual(args.length, 3, "expected 3 arguments - [address, unit256, log].");
+            capturedMints.push([...args])
+        });
+        for (let i = 0; i < 10; i++){
+            await contract.mint(BigNumber.from(`${i+1}`), { gasLimit: 300000 });
+        }
+        await sleep(20000);
+        contract.removeAllListeners();
+        assert.strictEqual(capturedMints.length > 9, true, "expected 10 captured events (Mint).")
```

Comment:
nit: instead of using `9` and `10` on line `336` we can declare another variable `const expectedEvents = 10` and use that